### PR TITLE
Create Special Metering Points

### PR DESCRIPTION
**Pull request by Korgath**
## Description

DOES NOT INCLUDE INTEGRATION EVENT

## References

* #0000

### CODE REVIEW

File: source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Special/SpecialMeteringPoint.cs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            };
+
+            return new BusinessRulesValidationResult(generalRuleCheckResult.Errors.Concat(rules.Where(r => r.IsBroken).Select(r => r.ValidationError).ToList()));
+        }
+
+        public static SpecialMeteringPoint Create(SpecialMeteringPointDetails meteringPointDetails)
+        {
+            if (!CanCreate(meteringPointDetails).Success)
+            {
+                throw new ConsumptionMeteringPointException($"Cannot create consumption metering point due to violation of one or more business rules.");
```
Comment (MrDach) [inline source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Special/SpecialMeteringPoint.cs:113]: Error message needs to be updated

**Review by MrDach — CHANGES_REQUESTED** [review]

File: source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Special/SpecialMeteringPoint.cs (hunk 1 above)
Comment (cmgrd) [reply source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Special/SpecialMeteringPoint.cs]: Fixed

**Review by MrDach — APPROVED** [review]

File: source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Special/Rules/MeterReadingOccurrenceConditionalRule.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            ValidationError = new InvalidMeterReadingOccurrenceRuleError(readingOccurrence.Name);
+        }
+
+        public bool IsBroken { get; }
+
+        public ValidationError ValidationError { get; }
+
+        private static bool ValidateRule(ReadingOccurrence readingOccurrence, MeteringPointType meteringPointType)
+        {
+            var t = meteringPointType.Name;
```
Comment (kft) [inline source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Special/Rules/MeterReadingOccurrenceConditionalRule.cs:37]: Do not seem to be used

File: source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Special/SpecialMeteringPoint.cs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+        public override void Connect(ConnectionDetails connectionDetails)
+        {
+            throw new NotImplementedException();
+        }
+
+        private static MeasurementUnitType GetUnitType(EnumerationType meteringPointType)
+        {
+            return meteringPointType == MeteringPointType.ExchangeReactiveEnergy ? MeasurementUnitType.KVArh : MeasurementUnitType.KWh;
+        }
```
Comment (kft) [inline source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Special/SpecialMeteringPoint.cs:144]: Maybe a factory method pr. metering point would be more intention revealing

**Review by cmgrd — APPROVED** [review]

**Review by kft — APPROVED** [review]
A couple of comment.

File: source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Special/Rules/MeterReadingOccurrenceConditionalRule.cs (hunk 2 above)
Comment (Korgath) [reply source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Special/Rules/MeterReadingOccurrenceConditionalRule.cs:37]: Will be handled in another PR

File: source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Special/SpecialMeteringPoint.cs (hunk 3 above)
Comment (Korgath) [reply source/Energinet.DataHub.MeteringPoints.Domain/MeteringPoints/Special/SpecialMeteringPoint.cs:144]: Handled in another PR

**Pull request merged**